EngineLayer/CommonParameters.cs

```diff
@@ -86,8 +89,11 @@ public CommonParameters()
         public double ScoreCutoff { get; private set; }
         public DigestionParams DigestionParams { get; private set; }
         public bool ReportAllAmbiguity { get; private set; }
-        public int TopNpeaks { get; private set; }
-        public double MinRatio { get; private set; }
+        public int? NumberOfPeaksToKeepPerWindow { get; private set; }
+        public double? MinimumAllowedIntensityRatioToBasePeak { get; private set; }
+        public double? WindowWidthThomsons { get; private set; }
+        public int? NumberOfWindows { get; private set; }
+        public bool NormalizePeaksAccrossAllWindows { get; private set; }
```

Comment:
I'm also worried about these being nullable - but I think they all have non-null "no filter" values that make sense.
```
NumberOfPeaksToKeepPerWindow = Infinity
MinimumAllowedIntensityRatioToBasePeak = 0
WindowWidthThomsons = Infinity
NumberOfWindows = 1
```

Does the .toml read/write still work with these values? And does the peak filtering (or lack of it, I guess) still work? And does it make sense in the context of peak filtering settings?

Reply:
I also don't really understand why there is both `WindowWidthThomsons` and `NumberOfWindows`; they seem to conflict with each other. Can you explain why there are two different settings that effectively set the width of the window? I might have asked that already but I forget the answer...